### PR TITLE
XDR - host_ip at incident alerts 

### PR DESCRIPTION
**Pull request by daryakoval**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31052

## Description
PaloAltoNetworksXDR.Incident.alerts.host_ip, when given host_ip list, in context we see it as comma separated string instead list of strings.

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CortexXDR/ReleaseNotes/2_4_13.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+- Added support for the **xdr-get-incident-extra-data** command which returns a list of host_ip's, at Incident Alerts.
```
Comment (ShahafBenYakir) [inline Packs/CortexXDR/ReleaseNotes/2_4_13.md:4]:
```suggestion
- Added support for host_ip's list in the data returned from **xdr-get-incident-extra-data** command.
```

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py [hunk 2]
```diff
@@ -966,14 +967,19 @@ def get_incident_extra_data_command(client, args):
...
+        del alert['host_ip']
     file_artifacts = raw_incident.get('file_artifacts').get('data')
     network_artifacts = raw_incident.get('network_artifacts').get('data')
 
     readable_output = [tableToMarkdown('Incident {}'.format(incident_id), incident)]
 
-    if len(alerts) > 0:
-        readable_output.append(tableToMarkdown('Alerts', alerts))
+    if len(hr_alerts) > 0:
+        readable_output.append(tableToMarkdown('Alerts', hr_alerts))
```
Comment (ShahafBenYakir) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:982]:
```suggestion
        readable_output.append(tableToMarkdown('Alerts', hr_alerts, headers=[key for key in hr_alerts[0] if key != 'host_ip'))
```

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py [hunk 3]
```diff
@@ -966,14 +967,19 @@ def get_incident_extra_data_command(client, args):
     incident = raw_incident.get('incident')
     incident_id = incident.get('incident_id')
     raw_alerts = raw_incident.get('alerts').get('data')
-    alerts = clear_trailing_whitespace(raw_alerts)
+    context_alerts = clear_trailing_whitespace(raw_alerts)
+    for alert in context_alerts:
+        alert['host_ip_list'] = alert.get('host_ip').split(',') if alert.get('host_ip') else []
+    hr_alerts = copy.deepcopy(context_alerts)
+    for alert in hr_alerts:
+        del alert['host_ip']
```
Comment (ShahafBenYakir) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:975]: Let's delete this

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py [hunk 4]
```diff
@@ -10,6 +10,7 @@
 import urllib3
 import traceback
 from operator import itemgetter
+import copy
```
Comment (ShahafBenYakir) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:13]: remove

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.yml [hunk 5]
```diff
@@ -335,7 +335,7 @@ script:
       type: String
     - contextPath: PaloAltoNetworksXDR.Incident.alerts.host_ip
```
Comment (ShahafBenYakir) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.yml:336]:
```suggestion
    - contextPath: PaloAltoNetworksXDR.Incident.alerts.host_ip_list
```

**Review by ShahafBenYakir — COMMENTED** [review]
- Please update the Readme to contain the new output instead of the old one(you can also do it manually)
- Go over the unit test and see if all changes are still required.

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py (hunk 3 above)
Comment (daryakoval) [reply Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:975]: done

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py [hunk 6]
```diff
@@ -966,14 +966,17 @@ def get_incident_extra_data_command(client, args):
...
     file_artifacts = raw_incident.get('file_artifacts').get('data')
     network_artifacts = raw_incident.get('network_artifacts').get('data')
 
     readable_output = [tableToMarkdown('Incident {}'.format(incident_id), incident)]
 
-    if len(alerts) > 0:
-        readable_output.append(tableToMarkdown('Alerts', alerts))
+    if len(context_alerts) > 0:
+        readable_output.append(tableToMarkdown('Alerts', context_alerts,
+                                               headers=[key for key in context_alerts[0] if key != 'host_ip']))
```
Comment (ShahafBenYakir) [inline Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:979]:
```suggestion
                                               headers=[key for key in context_alerts[0].keys() if key != 'host_ip']))
```

**Review by ShahafBenYakir — COMMENTED** [review]
Please tun the command and see that everything works

File: Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py (hunk 6 above)
Comment (daryakoval) [reply Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py:979]: when you do for on dict, you iterates on it's keys by default, without writing .keys() 
so it is working without the suggested change.

**Comment by daryakoval** [comment]
> Please tun the command and see that everything works

checked already before, and now again.. works :)

**Review by ShahafBenYakir — APPROVED** [review]

**Comment by Itay4** [comment]
@daryakoval please note the test fails in the build

**Pull request merged**